test_helpers.py

```diff
@@ -14,106 +14,119 @@
+    return get_until(
+        test_fixtures.PRIMARY_HOST,
+        test_fixtures.PRIMARY_PORT,
+        url,
+        requestHeaders,
+        predicate,
+        userAgent=userAgent)
+
 
 def get_url_until(url, requestHeaders, predicate, userAgent = None):
-  print "get_until(): %s" % url
-  timeout_seconds = time.time() + 5
-  ok = True
-  while True:
-    response, data = get_url(url, log = (ok==False), userAgent = userAgent, requestHeaders=requestHeaders)
-
-    if ok:
-      if predicate(response, data):
-        return response, data
-    else:
-      assert predicate(response, data)
-      return response, data
-
-    time.sleep(0.2)
-    if time.time() > timeout_seconds:
-      # TODO(oschaaf): doc about why we do this (print debug info)
-      if ok == False:
-        assert not "Timeout waiting for predicate"
-      ok = False
-
-# TODO(oschaaf): generic enough for all servers?
-def wait_untill_nginx_accepts():
-  timeout_seconds = time.time() + 5
-  while True:
-    try:
-      print "Wait until server is ready to handle requests"
-      response, data = get_primary("/")
-      print "Server ready"
-      return
-    except urllib3.exceptions.MaxRetryError as retry_err:
-      if isinstance(retry_err.reason, socket_error):
-        if retry_err.reason.errno != errno.ECONNREFUSED:
-          # Not the error we are looking for, re-raise
-          raise retry_err
-        # Connection still refused, sleep and try again.
+    print "get_until(): %s" % url
+    timeout_seconds = time.time() + 5
+    ok = True
+    while True:
+        response, data = get_url(
+            url, log=(ok == False),
+            userAgent=userAgent, requestHeaders=requestHeaders)
+
+        if ok:
+            if predicate(response, data):
+                return response, data
+        else:
+            assert predicate(response, data)
+            return response, data
+
+        time.sleep(0.1)
         if time.time() > timeout_seconds:
-          # TODO(oschaaf): raise something more specific
-          raise Exception("wait_untill_nginx_accepts timed out")
-        time.sleep(0.5)
-      else:
-        raise
+            # TODO(oschaaf): doc about why we do this (print debug info)
+            if not ok:
+                assert not "Timeout waiting for predicate"
+            ok = False
 
 # Helper to set up most filter tests.  Alternate between using:
 #  1) query-params vs request-headers
 #  2) ModPagespeed... vs PageSpeed...
 # to enable the filter so we know all combinations work.
-def filter_test(filter_name, filter_description, filter_spec_method):
-  # TODO(oschaaf): mentions other args the PageSpeedFilters=
-  url = "%s/%s.html?PageSpeedFilters=%s" % (test_fixtures.EXAMPLE_ROOT, filter_name, filter_name)
-  resp, body = get_primary(url)
-  return resp, body
-
-def http_date(d):
-  stamp = mktime(d.timetuple())
-  return format_date_time(stamp)
+def filter_test(filter_name, _filter_description, _filter_spec_method):
```

Comment:
> What about adding a test that tests all filters to see if they accept all types of query input [1] instead in favor of this? 

SGTM.

The test filter stuff in the shell scripts is also bad because some tests construct a url with query parameters and then happen to work because they're in a section where the test filter is using headers.  Then you add another test above and they break because now they're getting run with query params instead of headers.